features/admin.connections.v1/components/authenticator-grid.tsx

```diff
@@ -357,26 +336,34 @@ export const AuthenticatorGrid: FunctionComponent<AuthenticatorGridPropsInterfac
      * System defined authenticators cannot be deleted.
      */
     const handleLocalAuthenticatorDelete = async (authenticatorId: string): Promise<void> => {
-        connectedAppsOfLocalAuthenticator ? setIsConnectedAppsLoading(false) : setIsConnectedAppsLoading(true);
-
-        if (isLoadingConnectedAppsOfLocalAuthenticator || !connectedAppsOfLocalAuthenticator) {
-            return;
-        }
-
-        if (connectedAppsOfLocalAuthenticator?.count === 0) {
-            setDeletingIDP(
-                authenticators.find((idp: ConnectionInterface | AuthenticatorInterface) => idp.id === authenticatorId)
-            );
-            setShowDeleteConfirmationModal(true);
-        } else {
-            setIsConnectedAppsLoading(true);
-
-            setShowDeleteErrorDueToConnectedAppsModal(true);
-            loadConnectedAppDetails(connectedAppsOfLocalAuthenticator);
+        setIsConnectedAppsLoading(true);
 
-            setIsConnectedAppsLoading(false);
-            setShouldFetchLocalAuthenticatorConnectedApps(false);
-        };
+        getConnectedAppsOfAuthenticator(authenticatorId)
+            .then(async (response: ConnectedAppsInterface) => {
+                if (response.count === 0) {
+                    setDeletingIDP(
+                        authenticators.find(
+                            (idp: ConnectionInterface | AuthenticatorInterface) => idp.id === authenticatorId
```

Comment:
```suggestion
                            (idp: ConnectionInterface | AuthenticatorInterface) => idp.id === authenticatorId
```
Shall we use a genericc name rather than `idp` ?

Reply:
Fixed with 695d30b4d5d456e6778a32909bac704deee63353